### PR TITLE
corecfg: validate refresh.schedule when it is applied

### PR DESCRIPTION
**Pull request by mvo5**
Now that the configuration of the core snap is internal to snapd we can easily validate the refresh-schedule setting. This also uncovers that we set it to incorrect values in the test-suite which I removed now (the effect is the same, in both cases the default schedule is used).

### CODE REVIEW

File: corecfg/corecfg.go [hunk 1]
```diff
@@ -53,6 +53,9 @@ func Run(tr Conf) error {
 	if err := handleProxyStore(tr); err != nil {
 		return err
 	}
+	if err := validateRefreshSchedule(tr); err != nil {
```
Comment (pedronis) [inline corecfg/corecfg.go]: (nitpick)  should we call this "handle*" as the rest?  if we think is worth calling this "validate*" for clarity, then probably handleProxyStore should also be called validateProxyStore (it doesn't have side-effects as well)

Comment (mvo5) [reply corecfg/corecfg.go]: +1 for consistency. I slightly prefer to rename the calls that just validate so `valid{ProxyStore,RefreshSchedule}`. If that is ok with you I can do this in this PR. But no super strong opinion other than it should be both validate or both handle.

Comment (pedronis) [reply corecfg/corecfg.go]: than's fine here

**Comment by mvo5** [comment]
Thanks @pedronis - indeed, I updated the PR to have a working disable_refreshes again.

**Comment by pedronis** [comment]
@mvo5 thanks

**Pull request merged**